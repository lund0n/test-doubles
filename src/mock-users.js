export const ARTHUR_DENT = {
  id: 42,
  name: 'Arthur Dent',
  username: 'panickedBrit',
  email: 'DentedFord42@yahoo.com',
}

export const FORD_PREFECT = {
  id: 99,
  name: 'Ford Prefect',
  username: 'vogon-hater',
  email: 'TowelWearer1@gmail.com',
}
