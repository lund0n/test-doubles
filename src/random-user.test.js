import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import { ARTHUR_DENT, FORD_PREFECT } from './mock-users'
import RandomUser from './random-user'

test('Renders correct user', () => {
  const mockedUserPromise = Promise.resolve(ARTHUR_DENT)
  const getUser = jest.fn().mockReturnValue(mockedUserPromise)
  const component = renderer.create(<RandomUser getUser={getUser} />)
  return mockedUserPromise.then(() => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})

test('Renders new user on button click', () => {
  const user1 = Promise.resolve(ARTHUR_DENT)
  const user2 = Promise.resolve(FORD_PREFECT)
  const getUser = jest
    .fn()
    .mockReturnValueOnce(user1)
    .mockReturnValue(user2)
  const component = mount(<RandomUser getUser={getUser} />)
  return user1
    .then(() => {
      component.update() // Y U no work w/o an update here?
      expect(component.find('User').props()).toEqual(ARTHUR_DENT)
      component
        .update()
        .find('button')
        .simulate('click')
      return user2
    })
    .then(() => {
      component.update()
      expect(component.find('User').props()).toEqual(FORD_PREFECT)
    })
})
