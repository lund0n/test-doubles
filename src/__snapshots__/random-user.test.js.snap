// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correct user 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Get User
  </button>
  <dl>
    <dt>
      ID:
    </dt>
    <dd>
      42
    </dd>
    <dt>
      Name:
    </dt>
    <dd>
      Arthur Dent
    </dd>
    <dt>
      Username:
    </dt>
    <dd>
      panickedBrit
    </dd>
    <dt>
      Email:
    </dt>
    <dd>
      DentedFord42@yahoo.com
    </dd>
  </dl>
</div>
`;
